src/main/python/ttconv/vtt/writer.py

```diff
@@ -175,6 +181,16 @@ def process_p(self, region: ISD.Region, element: model.P, begin: Fraction, end:
         cue.set_line(round(position.v_offset.value + extent.height.value / 2))
         cue.set_align(VttCue.LineAlignment.center)
 
+    if self._config.text_align:
```

Comment:
I think you also need to take into consideration `tts:direction` if you want to resolve `start` and `end`:

https://github.com/sandflow/imscJS/blob/32d0767ba0830dbac984ff9584c45934b0b8cd45/src/main/js/html.js#L270